### PR TITLE
feat: add `CODEOWNERS` and refine `ISSUE TEMPLATE`

### PR DESCRIPTION
**Pull request by zmsn-2077**
## Description

Describe your changes in detail.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213
solve the feature request #232 
- [x] I have raised an issue to propose this change ([required](https://github.com/OmniSafeAI/omnisafe/issues) for new features and bug fixes)
feature request #232 

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/OmniSafeAI/omnisafe/blob/HEAD/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Watchers and contributors to Apache MXNet repo directories/packages/files
```
Comment (muchvo) [inline CODEOWNERS:1]:
```suggestion
# Watchers and contributors to Apache MXNet repo directories/packages/files
```
Why Apache MXNet?

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# Global owners
+*                                     @zmsn-2077 @Gaiejj @muchvo
+
+# Algorithms
+omnisafe/algorithms/model_based/      @hdadong
+omnisafe/algorithms/off_policy/       @Gaiejj
+omnisafe/algorithms/on_policy/        @Gaiejj @muchvo
+
+# Utils
+omnisafe/adapter/                     @Gaiejj @XuehaiPan
```
Comment (Gaiejj) [inline CODEOWNERS:19]: Maybe @rockmagma02 also own it

File: CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+omnisafe/models/                      @rockmagma02 @XuehaiPan
+omnisafe/utils/                       @Gaiejj @muchvo @XuehaiPan
+omnisafe/envs/                        @muchvo @XuehaiPan
```
[the remark this answers is not in the document]
Comment (zmsn-2077) [reply CODEOWNERS:22]: You can read line 6 in CODEOWNER.
# Anybody can add themselves or a team as additional watcher or contributor

**Review by Gaiejj — APPROVED** [review]
LGTM.

**Review by mickelliu — APPROVED** [review]
LGTM.

**Pull request merged**